src/filesys.cpp

```diff
@@ -382,19 +382,28 @@ std::string TempPath()
 
 #endif
 
-void GetRecursiveSubPaths(const std::string &path, std::vector<std::string> &dst, bool list_files)
+void GetRecursiveSubPaths(const std::string &path, std::vector<std::string> &dst, bool list_files, std::vector<char> ignore)
 {
 	std::vector<DirListNode> content = GetDirListing(path);
 	for (unsigned int i = 0; i<content.size(); i++) {
 		const DirListNode &n = content[i];
 		std::string fullpath = path + DIR_DELIM + n.name;
-
+		bool skip = false;
+		for (std::vector<char>::iterator i = ignore.begin();
+			i != ignore.end(); ++i) {
+			if (n.name.at(0) == *i) {
+				skip = true;
+			}
+		}
+		if (skip) {
+			continue;
```

Comment:
I'd prefer putting the if statements below into this rather than a continue

```cpp
if (!skip) {
    if (list_files || n.dir) 
         // ---
    if (n.dir)
         // ---
}
```